### DISCUSSION TITLE
renamerOnUpdate help

### DISCUSSION THREAD

**Discussion by MaltoxTv** [stashapp/stash#3435] [Q&A] [answered]
I believe I initially posted this in the wrong place and genuine bugs/etc should be posted there so i'm putting it here because I see similar posts. I am running Python 3.8, pip 23, and requests 2.28.2 so I should have all the prerequisites. I have Stash v0.18.0-42-gd80ec1d7 installed into a subfolder in my downloads folder, a plugin folder within that, and then finally the renamer script. Stash recognizes the script but won't even blurt out a log file let alone actually do the renaming and moving I spent hours trying to set it up to do. I've spent 3 days trying to figure this out myself but am getting nowhere, I would highly appreciate some help! I've attached my config file, hopefully with that, my stash directory info, and what programs/versions i'm using someone can help identify what I did wrong. Could be a simple syntax error honestly, I don't know what i'm doing or if i'm even allowed to request help here like this

I have also included a list of my current tags and the directories i'm trying to put them in. I did try to run this without the $studio_hierarchy addition when moving the videos per tags without luck either. This is a VR porn library that I'm trying to use with DeoVR so having the folder physically organized by tags, studios, and performers would be very helpful. Is there a way to get this script to create shortcuts to the videos? For instance, many videos have multiple tags so if I create a folder for every single tag then many videos will be referenced multiple times, so where does it go? Into the performer folder and a shortcut is placed in all folders that reference it. I think I have a fundamental misunderstanding of how this works...
[TADCS.txt](https://github.com/stashapp/stash/files/10726490/TADCS.txt)
[Taglist2_12_2023CS.txt](https://github.com/st
[RenamerOnUpdate_configexample.txt](https://github.com/stashapp/stash/files/10726491/RenamerOnUpdate_configexample.txt)
ashapp/stash/files/10726489/Taglist2_12_2023CS.txt)

**Comment by bnkai** [comment]
Some troubleshooting info first
- Make sure you have python 3.10 ( 3.8 may or may not be ok )
- Set stash's log level to debug (enabling the file log as well is a good idea)
- Have a look at the logs / terminal output when the plugin is run to see if something is printed that can help us some more. Without any more info we cant be sure what is going on.

From a quick look in your config  file
- You have some of the options twice (first )? copy paste error ?
- `only_organized = True` that means that only scenes that you have set as organized in stash will be renamed
- I would advice to start with simple things first before moving to more complex. Set `dry_run` to `True` and use the simpler `tag_templates` to see if anything works/printed out when you update a scene. If that works then try editing some of the path moving options

>Is there a way to get this script to create shortcuts to the videos?

No the script only renames/moves the file. If there are multiple matches the file will `probably` be renamed according to the first

**Reply by MaltoxTv** [reply]
https://github.com/stashapp/CommunityScripts/tree/main/plugins/renamerOnUpdate states "Python 3.6+ (Tested on Python v3.9.1 64bit, Win10)" so I assumed I was safe with 3.8, my next step was to try it again with Python 3.9.1 since I'm also running windows 10 64bit. I'll give 3.10 a shot first if the below doesn't lead to a solve. I sincerely appreciate the tip about stash's log level, once I ran a "rename all scenes" I discovered this in there: 
"023-02-13 17:39:22
Error   
Error running plugin task: error running plugin: exec: "plugins\\renamerOnUpdate\\plugins\\renamerOnUpdate\\python": file does not exist" which leads me to believe I simply don't have the stash directory and plugins configured properly. That's why I mentioned I had everything in a subdirectory of my downloads folder lol, I thought it might cause an issue. Do I need to reinstall and use stash's recommended location settings?

You guessed right, the duplicate only_organized was a copy and paste error when putting it into the txt. I've been trying to get it running with one tag as the moving trigger and only_organized set to the default false too, fortunately it seems like i'm getting closer to finding the solution given the error in the stash log. Thank you for your help man, genuinely. I'm ignorant

**Reply by bnkai** [reply]
The error you pasted means that you havent setup python correctly (edit the .yml file and try using `python3` instead of `python`, reload plugins and retry to see if it makes a difference). 
If you used the windows store version then remove that and try the official one (3.10+) from python.org  eg <https://phoenixnap.com/kb/how-to-install-python-3-windows>, making sure that the `PATH`  option is ticked.
If possible join stash's discord server as it will be easier(and probably faster) to troubleshoot the issue there

**Reply by MaltoxTv** [reply]
Okay, Python 3.10.10 installed and path option ticked. Requests installed, .yml edited to python3 instead of python and plugins reloaded. Unfortunately, I got the below error:
"Error running plugin task: error running plugin: exec: "plugins\\renamerOnUpdate\\python3": file does not exist"

When I undid the yml change I still get this error :(
"Error running plugin task: error running plugin: exec: "plugins\\renamerOnUpdate\\python": file does not exist"

What's the discord channel? I'll hop in

**Reply by DogmaDragon** [reply]
Try restarting the host machine after the new Python installation. 

Community Discord: <https://discord.gg/frQ7qBvB3S>

**Reply by bnkai** [reply]
As mentioned above
Did you restart stash after installing python, it needs the restart to get the updated PATH env variable

**Reply by MaltoxTv** [reply]
I restarted and ran it, it created and deleted the organized file multiple times and in the process deleted a lot of videos. I have backups when playing with such things for a reason luckily. Here's another thing I found in the stash log: 
"[Plugin / renamerOnUpdate] error when trying to update the database (You need to setup a library with the new location (D:\CVRPO\Unorganized) and scan at least 1 file), revert the move..."

I've attached the stash log and the one generated by the plugin
[rename_log.txt](https://github.com/stashapp/stash/files/10727436/rename_log.txt)
[Stashlog.txt](https://github.com/stashapp/stash/files/10727439/Stashlog.txt)

I should probably mention that I already had an empty file create in that directory name Unorganized file that it deleted, recreated, added to, deleted, etc. I am in the discord now :)

**Comment by bnkai** [comment, marked as answer]
- you didnt set `dry_run` to `True` ? 
- as the log says to move to a new folder that folder needs to be in stash's library so add it there and do a new scan 
- the above means that the plugin runs and you just need to fine tweak your config options
- for more info post in the help or plugins-and-scripts channel, not sure what your discord usename is (Maltox?)

**Reply by MaltoxTv** [reply]
It is Maltox, and no sir I forgot to do that and i'm lucky it isn't as bad as it could have been. I'll move to the discord, and thanks for the information. Now I know I need to have these be created and added to Stash before it's run
